diff -ruN rabbitmq-server-2.5.0-old/Makefile rabbitmq-server-2.5.0-new/Makefile
--- rabbitmq-server-2.5.0-old/Makefile	2011-06-09 15:24:13.000000000 +0100
+++ rabbitmq-server-2.5.0-new/Makefile	2012-08-23 13:26:12.030895537 +0100
@@ -97,7 +97,7 @@
 	escript generate_app $< $@ $(SOURCE_DIR)
 
 $(EBIN_DIR)/%.beam: $(SOURCE_DIR)/%.erl | $(DEPS_FILE)
-	erlc $(ERLC_OPTS) -pa $(EBIN_DIR) $<
+	$(ERLC) $(ERLC_OPTS) -pa $(EBIN_DIR) $<
 
 $(INCLUDE_DIR)/rabbit_framing.hrl: codegen.py $(AMQP_CODEGEN_DIR)/amqp_codegen.py $(AMQP_SPEC_JSON_FILES_0_9_1) $(AMQP_SPEC_JSON_FILES_0_8)
 	$(PYTHON) codegen.py --ignore-conflicts header $(AMQP_SPEC_JSON_FILES_0_9_1) $(AMQP_SPEC_JSON_FILES_0_8) $@
@@ -233,9 +233,9 @@
 
 # xmlto can not read from standard input, so we mess with a tmp file.
 %.gz: %.xml $(DOCS_DIR)/examples-to-end.xsl
-	xmlto --version | grep -E '^xmlto version 0\.0\.([0-9]|1[1-8])$$' >/dev/null || opt='--stringparam man.indent.verbatims=0' ; \
-	    xsltproc --novalid $(DOCS_DIR)/examples-to-end.xsl $< > $<.tmp && \
-	    xmlto -o $(DOCS_DIR) $$opt man $<.tmp && \
+	$(XMLTO) --version | grep -E '^xmlto version 0\.0\.([0-9]|1[1-8])$$' >/dev/null || opt='--stringparam man.indent.verbatims=0' ; \
+	    $(XSLTPROC) --novalid $(DOCS_DIR)/examples-to-end.xsl $< > $<.tmp && \
+	    $(XMLTO) -o $(DOCS_DIR) $$opt man $<.tmp && \
 	    gzip -f $(DOCS_DIR)/`basename $< .xml`
 	rm -f $<.tmp
 
@@ -243,7 +243,7 @@
 # Do not fold the cp into previous line, it's there to stop the file being
 # generated but empty if we fail
 $(SOURCE_DIR)/%_usage.erl:
-	xsltproc --novalid --stringparam modulename "`basename $@ .erl`" \
+	$(XSLTPROC) --novalid --stringparam modulename "`basename $@ .erl`" \
 		$(DOCS_DIR)/usage.xsl $< > $@.tmp
 	sed -e 's/"/\\"/g' -e 's/%QUOTE%/"/g' $@.tmp > $@.tmp2
 	fold -s $@.tmp2 > $@.tmp3
@@ -254,10 +254,10 @@
 # the file to make internal links.
 %.man.xml: %.xml $(DOCS_DIR)/html-to-website-xml.xsl
 	cp $< `basename $< .xml`.xml && \
-		xmlto xhtml-nochunks `basename $< .xml`.xml ; rm `basename $< .xml`.xml
+		$(XMLTO) xhtml-nochunks `basename $< .xml`.xml ; rm `basename $< .xml`.xml
 	cat `basename $< .xml`.html | \
-	    xsltproc --novalid $(DOCS_DIR)/remove-namespaces.xsl - | \
-		xsltproc --novalid --stringparam original `basename $<` $(DOCS_DIR)/html-to-website-xml.xsl - | \
+	    $(XSLTPROC) --novalid $(DOCS_DIR)/remove-namespaces.xsl - | \
+		$(XSLTPROC) --novalid --stringparam original `basename $<` $(DOCS_DIR)/html-to-website-xml.xsl - | \
 		xmllint --format - > $@
 	rm `basename $< .xml`.html
 
@@ -276,13 +276,14 @@
 	mkdir -p $(TARGET_DIR)/plugins
 	echo Put your .ez plugin files in this directory. > $(TARGET_DIR)/plugins/README
 
-install_docs: docs_all install_dirs
-	for section in 1 5; do \
-		mkdir -p $(MAN_DIR)/man$$section; \
-		for manpage in $(DOCS_DIR)/*.$$section.gz; do \
-			cp $$manpage $(MAN_DIR)/man$$section; \
-		done; \
-	done
+install_docs:
+	#docs_all install_dirs
+	#for section in 1 5; do \
+	#	mkdir -p $(MAN_DIR)/man$$section; \
+	#	for manpage in $(DOCS_DIR)/*.$$section.gz; do \
+	#		cp $$manpage $(MAN_DIR)/man$$section; \
+	#	done; \
+	#done
 
 install_dirs:
 	@ OK=true && \
